internal/export/distro/registrations.go

```diff
@@ -187,30 +208,32 @@ func (reg registrationInfo) processEvent(event *models.Event) {
 		compressed = reg.compression.Transform(formatted)
 	}
 
-	encrypted := compressed
+	bytes := compressed
 	if reg.encrypt != nil {
-		encrypted = reg.encrypt.Transform(compressed)
+		bytes = reg.encrypt.Transform(compressed)
 	}
-
-	if reg.sender.Send(encrypted, event) && Configuration.Writable.MarkPushed {
-		id := event.ID
-		err := ec.MarkPushed(id, context.Background())
-
-		if err != nil {
-			LoggingClient.Error(fmt.Sprintf("Failed to mark event as pushed : event ID = %s: %s", id, err))
-		}
+	if reg.sender.Send(bytes, ctx) && Configuration.Writable.MarkPushed {
+		return ec.MarkPushed(event.ID, ctx)
```

Comment:
Should we add comment here for consistency with handleCBOR? 
`// JSON content type is not needed when calling back to core-data`

Reply:
The JSON ContentType is added by the EventClient `ec` if one isn't already specified in the context.

Reply:
Ah got it, thanks for the explanation.